src/psyclone/psyir/backend/visitor.py

```diff
@@ -253,9 +253,16 @@ def _visit(self, node):
 
                 # Add preceding comment if available
                 if isinstance(node, CommentableMixin):
-                    if node.preceding_comment and self._COMMENT_PREFIX:
-                        result += (self._nindent + self._COMMENT_PREFIX +
-                                   node.preceding_comment + "\n")
+                    # and is in a location that allows line comments
+                    valid_locations = (Schedule, Container)
+                    if not (node.parent and
+                            not isinstance(node.parent, valid_locations)):
+                        if node.preceding_comment and self._COMMENT_PREFIX:
+                            lines = node.preceding_comment.split('\n')
+                            for line in lines:
```

Comment:
Please could you update https://psyclone-dev.readthedocs.io/en/latest/psyir.html#comments-attached-to-psyir-nodes to describe this support for multi-line comments.

Reply:
I think this was more a bugfix and it now does the expected behaviour of adding the comment prefix for each line break. I just updated the example to show that this is possible, but I am not sure if this is what you meant.

Reply:
I think the update to the example is sufficient - thanks :-)